### PR TITLE
Only compile Terraform's versioned files once per run on MM.

### PR DESCRIPTION
**Pull request by rileykarson**
Most compiled files aren't provider-specific, so we only need to compile them once. #647 was a great idea and brought compile times on my machine running `bundle exec compiler -a -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/" -v ga` from ~47s to ~39s.

Let's go even further and bring them to 23s 🙂 

-----------------------------------------------------------------
# [all]
no-op

### CODE REVIEW

File: products/storage/ansible.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ overrides: !ruby/object:Provider::ResourceOverrides
 examples: !ruby/object:Api::Resource::HashArray
 files: !ruby/object:Provider::Config::Files
   compile:
-<%= lines(indent(compile('provider/ansible/common~compile.yaml'), 4)) -%>
+<%= lines(indent(compile('provider/ansible/product~compile.yaml'), 4)) -%>
```
Comment (chrisst) [inline products/storage/ansible.yaml]: It looks like all the ansible products are using a common~compile step, so shouldn't you be able to just remove these lines and let the common logic kick in rather then renaming them to product compile? I also don't see `provider/ansible/product~compile.yaml` in this PR

File: provider/core.rb [hunk 2]
```diff
@@ -97,11 +97,21 @@ def copy_files(output_folder)
     def copy_common_files(output_folder, _version_name = nil)
       provider_name = self.class.name.split('::').last.downcase
       return unless File.exist?("provider/#{provider_name}/common~copy.yaml")
-      files = YAML.safe_load(compile("provider/#{provider_name}/common~copy.yaml"))
 
+      Google::LOGGER.info "Copying common files for #{provider_name}"
```
[the remark this answers is not in the document]
Comment (rileykarson) [reply provider/core.rb]: Imo this should be at the same level as the product compilations such as `Compiling 'products/resourcemanager' output to`, and they're at info.

File: products/storage/ansible.yaml (hunk 1 above)
Comment (rileykarson) [reply products/storage/ansible.yaml]: It's in there! It's just a rename so it's easy to miss on GitHub.

The Ansible compiled files are all compiled with product-specific attributes afaik - so `product~compile` is a better name now given how we've changed TF.

File: provider/core.rb [hunk 3]
```diff
@@ -97,11 +97,21 @@ def copy_files(output_folder)
...
+      files = YAML.safe_load(compile("provider/#{provider_name}/common~copy.yaml"))
       copy_file_list(output_folder, files)
     end
 
+    def compile_common_files(output_folder, version_name = nil)
+      provider_name = self.class.name.split('::').last.downcase
```
[the remark this answers is not in the document]
Comment (rileykarson) [reply provider/core.rb]: Sure!

Since I can't find documentation for `compile` anywhere, I commented above `compile_files`.

**Pull request merged**